src/main/java/org/jahia/modules/serversettings/flow/SiteBean.java

```diff
@@ -230,85 +233,86 @@ public void setEditModules(boolean editModules) {
+                        .code("serverSettings.manageWebProjects.warningMsg.noTemplateSets")
+                        .build());
             }
 
-            if (title != null && (title.length() > 0) && serverName != null && (serverName.length() > 0)
-                    && siteKey != null && (siteKey.length() > 0)) {
-                if (!sitesService.isSiteKeyValid(siteKey)) {
-                    messages.addMessage(new MessageBuilder().error().source("siteKey")
-                            .code("serverSettings.manageWebProjects.warningMsg.onlyLettersDigitsUnderscore").build());
+            String userSiteKey = (String) context.getUserValue(SITE_KEY_FIELD);
+            String userTitle = (String) context.getUserValue(TITLE_FIELD);
+            String userServerName = (String) context.getUserValue(SERVER_NAME_FIELD);
+            if (StringUtils.isNotBlank(userTitle) && StringUtils.isNotBlank(userServerName) && StringUtils.isNotBlank(userSiteKey)) {
+                JahiaSitesService sitesService = ServicesRegistry.getInstance().getJahiaSitesService();
+
+                if (!sitesService.isSiteKeyValid(userSiteKey)) {
+                    messages.addMessage(new MessageBuilder()
+                            .error()
+                            .source(SITE_KEY_FIELD)
+                            .code("serverSettings.manageWebProjects.warningMsg.onlyLettersDigitsUnderscore")
+                            .build());
                 }
 
-                testServerNames(messages, sitesService);
+                testServerNames(messages);
 
-                if (sitesService.getSiteByKey(siteKey) != null) {
+                if (sitesService.getSiteByKey(userSiteKey) != null) {
                     messages.addMessage(new MessageBuilder()
                             .error()
-                            .source("siteKey")
-                            .code("serverSettings.manageWebProjects.warningMsg.chooseAnotherSiteKey").build());
+                            .source(SITE_KEY_FIELD)
+                            .code("serverSettings.manageWebProjects.warningMsg.chooseAnotherSiteKey")
+                            .build());
                 }
             } else {
                 messages.addMessage(new MessageBuilder()
                         .error()
-                        .source("siteKey")
-                        .code("serverSettings.manageWebProjects.warningMsg.completeRequestInfo").build());
+                        .source(SITE_KEY_FIELD)
+                        .code("serverSettings.manageWebProjects.warningMsg.completeRequestInfo")
+                        .build());
             }
         } catch (JahiaException e) {
             logger.error(e.getMessage(), e);
         }
     }
 
     public void validateCreateSiteSelectModules(ValidationContext context) {
-        MessageContext messages = context.getMessageContext();
+        // Fix an issue with an empty modules value
```

Comment:
Comment should clarify what this is actually fixing. "List of modules is not being cleared when all modules have been unselected" 'with a reference to the bug this is fixing ?